hq/app/aws/cloudformation/CloudFormation.scala

```diff
@@ -1,52 +1,52 @@
-import com.amazonaws.services.cloudformation.{AmazonCloudFormationAsync, AmazonCloudFormationAsyncClientBuilder}
+import com.amazonaws.services.ec2.AmazonEC2Async
 import model.{AwsAccount, AwsStack}
-import utils.attempt.Attempt
+import utils.attempt.{Attempt, FailedAttempt, Failure}
 
 import scala.collection.JavaConverters._
 import scala.concurrent.ExecutionContext
 
 object CloudFormation {
-  private def client(auth: AWSCredentialsProviderChain, region: Region): AmazonCloudFormationAsync = {
-    AmazonCloudFormationAsyncClientBuilder.standard()
-      .withCredentials(auth)
-      .withRegion(region.getName)
-      .build()
-  }
-  private def client(awsAccount: AwsAccount, region: Region): AmazonCloudFormationAsync = {
-    val auth = AWS.credentialsProvider(awsAccount)
-    client(auth, region)
-  }
 
+  def client(cfnClients: Map[(String, Regions),AmazonCloudFormationAsync], awsAccount: AwsAccount, region: Regions): Attempt[AmazonCloudFormationAsync] =
+    Attempt.fromOption(cfnClients.get((awsAccount.id, region)), FailedAttempt(Failure(
+      s"No AWS Cloudformation Client exists for ${awsAccount.id} and $region",
+      s"Cannot find Cloudformation Client",
+      500
+    )
+  ))
   private def getStackDescriptions(client: AmazonCloudFormationAsync)(implicit ec: ExecutionContext): Attempt[List[Stack]] = {
     val request = new DescribeStacksRequest()
     handleAWSErrs(awsToScala(client.describeStacksAsync)(request)).map(_.getStacks.asScala.toList)
   }
 
-  private def getStacks(account: AwsAccount, region: Region)(implicit ec: ExecutionContext): Attempt[List[AwsStack]] = {
-    val cloudClient = CloudFormation.client(account, region)
+  private def getStacks(account: AwsAccount, region: Regions, cfnClients: Map[(String, Regions),AmazonCloudFormationAsync])(implicit ec: ExecutionContext): Attempt[List[AwsStack]] = {
     for {
+      cloudClient <- client(cfnClients, account, region)
       stacks <- getStackDescriptions(cloudClient)
     } yield parseStacks(stacks, region)
   }
 
-  def getStacksFromAllRegions(account: AwsAccount)(implicit ec: ExecutionContext): Attempt[List[AwsStack]] = {
-    val regionClient = EC2.client(account)
+  def getStacksFromAllRegions(
+    account: AwsAccount,
+    cfnClients: Map[(String, Regions), AmazonCloudFormationAsync],
+    ec2Clients: Map[(String, Regions), AmazonEC2Async]
+  )(implicit ec: ExecutionContext): Attempt[List[AwsStack]] = {
     for {
+      regionClient <- EC2.client(ec2Clients, account, Regions.EU_WEST_1)
       availableRegions <- EC2.getAvailableRegions(regionClient)
-      regions = availableRegions.map(region => Region.getRegion(Regions.fromName(region.getRegionName)))
-      stacks <- Attempt.flatTraverse(regions)(region => getStacks(account, region))
+      regions = availableRegions.map(region => Regions.fromName(region.getRegionName))
+      stacks <- Attempt.flatTraverse(regions)(region => getStacks(account, region, cfnClients))
```

Comment:
The AWS SDK `Regions.values()` will get you all the regions but these are not necessarily ones that you or the application will have access to. 😅 

This was fun to discover, since `availableRegions <- EC2.getAvailableRegions(regionClient)` was added to work around the existence of `GovCloud("us-gov-west-1")` which returns **access denied** errors for some strange reason (what could that be?) when trying to `describeResources` for the region. 
🙈 🙉 🙊

So you could make clients for all regions except the US GovCloud, then hopefully all of the clients will have the required permissions to make the API calls the application wants to.

I feel like I have notes on this somewhere (which I will try to find) so this may not or not longer apply. 🙃